packages/x-components/src/x-modules/facets/components/facets/facets-provider.vue

```diff
@@ -19,97 +16,107 @@
    *
    * @public
    */
-  @Component({
-    mixins: [xComponentMixin(facetsXModule)]
-  })
-  export default class FacetsProvider extends Vue {
-    /**
-     * An facet group identifier to distinguish the provided facets from other facets like the
-     * `Search X-Module` facets.
-     *
-     * @public
-     */
-    @Prop({ default: 'provided-facets' })
-    public groupId!: GroupId;
-
-    /**
-     * The facets to provide to the `Facets X-Module` state. They have to include the
-     * {@link @empathyco/x-types#Filter}.
-     *
-     * @internal
-     */
-    @Prop({ required: true })
-    public facets!: Facet[];
-
-    /**
-     * Temporarily stores the selected filters from the {@link FacetsProvider.facets} prop.
-     * This is necessary to handle the {@link FacetsXEvents.UserChangedSelectedFilters} event.
-     *
-     * @internal
-     */
-    protected selectedFilters: Filter[] | null = null;
-
-    /**
-     * A computed property to group the facets and the groupId. This is used by the watcher.
-     *
-     * @returns The FacetGroup with the facets and the group id.
-     *
-     * @internal
-     */
-    protected get facetsGroup(): FacetsGroup {
-      return { id: this.groupId, facets: this.facets };
-    }
-
-    /**
-     * Emits the {@link FacetsXEvents.UserChangedSelectedFilters} event when the user changes
-     * the selected filters.
-     *
-     * @param selectedFilters - The new list of selected filters.
-     * @internal
-     */
-    @XOn('SelectedFiltersChanged')
-    emitSelectedFiltersChanged(selectedFilters: Filter[]): void {
-      if (
-        this.selectedFilters === null ||
-        areFiltersDifferent(this.selectedFilters, selectedFilters)
-      ) {
-        this.$x.emit('UserChangedSelectedFilters', selectedFilters);
+  export default defineComponent({
+    name: 'FacetsProvider',
+    xModule: facetsXModule.name,
```

Comment:
I think you should register the module in the `x-module.ts` file.

You can register it with `XPlugin.registerXModule(facetsXModule);`